docs/utilities/batch.md

```diff
@@ -1,460 +1,322 @@
+    }
+    RecordHandler --> BatchProcessor: Collect results
+    BatchProcessor --> LambdaResponse: Report items that failed processing
+```
 
 **Key Features**
 
-* Prevent successfully processed messages from being returned to SQS
-* A simple interface for individually processing messages from a batch
+
+* Reports batch item failures to reduce number of retries for a record upon errors
+* Simple interface to process each batch record
+* Integrates with Java Events library and the deserialization module 
+* Build your own batch processor by extending primitives
 
 **Background**
 
-When using SQS as a Lambda event source mapping, Lambda functions can be triggered with a batch of messages from SQS. 
-If your function fails to process any message from the batch, the entire batch returns to your SQS queue, and your 
-Lambda function will be triggered with the same batch again. With this utility, messages within a batch will be handled individually - only messages that were not successfully processed
-are returned to the queue.
+When using SQS, Kinesis Data Streams, or DynamoDB Streams as a Lambda event source, your Lambda functions are 
+triggered with a batch of messages.
+If your function fails to process any message from the batch, the entire batch returns to your queue or stream.
+This same batch is then retried until either condition happens first:
+**a)** your Lambda function returns a successful response, 
+**b)** record reaches maximum retry attempts, or 
+**c)** records expire.
+
+```mermaid
+journey
+  section Conditions
+    Successful response: 5: Success
+    Maximum retries: 3: Failure
+    Records expired: 1: Failure
+```
+
+This behavior changes when you enable Report Batch Item Failures feature in your Lambda function event source configuration:
+
+<!-- markdownlint-disable MD013 -->
+* [**SQS queues**](#sqs-standard). Only messages reported as failure will return to the queue for a retry, while successful ones will be deleted.
+  * [**Kinesis data streams**](#kinesis-and-dynamodb-streams) and [**DynamoDB streams**](#kinesis-and-dynamodb-streams).
+  Single reported failure will use its sequence number as the stream checkpoint. 
+  Multiple  reported failures will use the lowest sequence number as checkpoint.
+
+With this utility, batch records are processed individually – only messages that failed to be processed 
+return to the queue or stream for a further retry. You simply build a `BatchProcessor` in your handler,
+and return its response from the handler's `processMessage` implementation. Exceptions are handled 
+internally and an appropriate partial response for the message source is returned to Lambda for you.
 
 !!! warning
-    While this utility lowers the chance of processing messages more than once, it is not guaranteed. We recommend implementing processing logic in an idempotent manner wherever possible.
+    While this utility lowers the chance of processing messages more than once, but it is not guaranteed. 
```

Comment:
you're the english guy but I'm not sure about the "but" in this sentence....

Reply:
Fixed!